packages/core/src/external/carequality/ihe-gateway-v2/ihe-gateway-v2-logic.ts

```diff
@@ -105,20 +145,14 @@ export async function createSignSendProcessXCPDRequest({
 }): Promise<void> {
   const log = getLog("createSignSendProcessXCPDRequest");
   const signedRequests = createAndSignBulkXCPDRequests(xcpdRequest, samlCertsAndKeys);
-  const responses = await sendSignedXCPDRequests({
-    signedRequests,
-    samlCertsAndKeys,
-    patientId,
-    cxId,
-  });
-  const results: OutboundPatientDiscoveryResp[] = responses.map(response => {
-    return processXCPDResponse({
-      xcpdResponse: response,
+  const resultPromises = signedRequests.map(async (signedRequest, index) => {
+    const result = await sendProcessRetryXcpdRequest({
+      signedRequest,
+      samlCertsAndKeys,
       patientId,
       cxId,
+      index,
     });
-  });
-  for (const result of results) {
     try {
       // TODO not sure if we should retry on timeout
       await executeWithNetworkRetries(async () => axios.post(pdResponseUrl, result), { log });
```

Comment:
Why are we awaiting this?

Reply:
Anything that runs on a lambda needs to await, otherwise it can be lost - the lambda env kills the Node process when the `handle` fn returns.

Reply:
Right -- forgot this is running on lambda. 👍